### PR TITLE
Implement analog stick range modifier

### PR DESCRIPTION
**Pull request by MutantAura**
Adds range modification to the analog sticks of a given controller. Partially addresses #2059.

Unlike an older PR (#2095) this PR will not support the addition of a keyboard modifier button directly (potentially a follow-up). This is a simple float multiplication modifier of the stick vector applied before circular clamping in order to maintain a natural min and max of the analog values.

Values above 1.00 = Easier to reach max input at lower analog distance.
Values below 1.00 = Harder/Impossible to reach max input (this will reduce maximum input).
This is in-line with Cemu's convention and so should be familiar to users.

Default values: 
image

1.00 modification:
image

1.50 modification:
image

0.50 modification: 
image

### CODE REVIEW

File: Ryujinx.Input/HLE/NpadController.cs [hunk 1]
```diff
@@ -412,9 +412,9 @@ private static short ClampAxis(float value)
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static JoystickPosition ClampToCircle(JoystickPosition position)
+        private static JoystickPosition ClampToCircle(JoystickPosition position, float range)
         {
-            Vector2 point = new Vector2(position.Dx, position.Dy);
+            Vector2 point = new Vector2(position.Dx, position.Dy)*range;
```
Comment (AcK77) [inline Ryujinx.Input/HLE/NpadController.cs:417]:
```suggestion
            Vector2 point = new Vector2(position.Dx, position.Dy) * range;
```

File: Ryujinx/Configuration/ConfigurationFileFormat.cs [hunk 2]
```diff
@@ -14,7 +14,7 @@ public class ConfigurationFileFormat
         /// <summary>
         /// The current version of the file format
         /// </summary>
-        public const int CurrentVersion = 32;
+        public const int CurrentVersion = 33;
```
Comment (AcK77) [inline Ryujinx/Configuration/ConfigurationFileFormat.cs:17]: Version needs to be updated

File: Ryujinx/Configuration/ConfigurationState.cs [hunk 3]
```diff
@@ -925,7 +925,21 @@ public void Load(ConfigurationFileFormat configurationFileFormat, string configu
                     ShowUi = configurationFileFormat.Hotkeys.ShowUi,
                     Pause = Key.F5
                 };
+            }
+
+            if (configurationFileFormat.Version < 33)
```
Comment (AcK77) [inline Ryujinx/Configuration/ConfigurationState.cs:930]: Version needs to be updated

File: Ryujinx/Configuration/ConfigurationState.cs [hunk 4]
```diff
@@ -925,7 +925,21 @@ public void Load(ConfigurationFileFormat configurationFileFormat, string configu
...
+            {
+                Common.Logging.Logger.Warning?.Print(LogClass.Application, $"Outdated configuration version {configurationFileFormat.Version}, migrating to version 33.");
```
Comment (AcK77) [inline Ryujinx/Configuration/ConfigurationState.cs:932]: Version needs to be updated

**Review by AcK77 — CHANGES_REQUESTED** [review]
Ryujinx/_schema.json doesn’t exist anymore, it’s needs a rebase too.

File: Ryujinx.Input/HLE/NpadController.cs (hunk 1 above)
Comment (MutantAura) [reply Ryujinx.Input/HLE/NpadController.cs]: updated

File: Ryujinx/Configuration/ConfigurationFileFormat.cs (hunk 2 above)
Comment (MutantAura) [reply Ryujinx/Configuration/ConfigurationFileFormat.cs]: all config versions updated to 35

**Review by AcK77 — APPROVED** [review]
Lgtm! Thanks!

File: Ryujinx/Ui/Windows/ControllerWindow.cs [hunk 5]
```diff
@@ -1013,6 +1025,8 @@ private void ProfileLoad_Activated(object sender, EventArgs args)
                         ControllerType   = ControllerType.JoyconPair,
                         DeadzoneLeft     = 0.1f,
                         DeadzoneRight    = 0.1f,
+                        RangeLeft        = 1.0f,
+                        RangeRight       = 1.0f,
```
Comment (Thog) [inline Ryujinx/Ui/Windows/ControllerWindow.cs:1029]: This needs to be also done on profile json load as the value will end up being 0.0 if the field isn't present

**Review by Thog — CHANGES_REQUESTED** [review]

File: Ryujinx/Ui/Windows/ControllerWindow.cs (hunk 5 above)
Comment (MutantAura) [reply Ryujinx/Ui/Windows/ControllerWindow.cs:1038]: Workaround added?

File: Ryujinx/Ui/Windows/ControllerWindow.cs [hunk 6]
```diff
@@ -510,12 +518,23 @@ private void SetValues(InputConfig config)
...
                     _controllerDeadzoneRight.Value    = controllerConfig.DeadzoneRight;
+                    _controllerRangeLeft.Value        = controllerConfig.RangeLeft;
+                    _controllerRangeRight.Value       = controllerConfig.RangeRight;
                     _controllerTriggerThreshold.Value = controllerConfig.TriggerThreshold;
                     _sensitivity.Value                = controllerConfig.Motion.Sensitivity;
                     _gyroDeadzone.Value               = controllerConfig.Motion.GyroDeadzone;
                     _enableMotion.Active              = controllerConfig.Motion.EnableMotion;
                     _enableCemuHook.Active            = controllerConfig.Motion.MotionBackend == MotionInputBackendType.CemuHook;
 
+                    //If both stick ranges are 0 (usually indicative of an outdated profile load) then both sticks will be set to 1.0
```
Comment (AcK77) [inline Ryujinx/Ui/Windows/ControllerWindow.cs:529]:
```suggestion
                    // If both stick ranges are 0 (usually indicative of an outdated profile load) then both sticks will be set to 1.0.
```

Comment (MutantAura) [reply Ryujinx/Ui/Windows/ControllerWindow.cs]: updated

**Review by gdkchan — APPROVED** [review]
Looks fine.

**Review by Thog — APPROVED** [review]
lgtm

**Pull request merged**